x/warp/README.md

```diff
@@ -142,6 +91,30 @@ Since the predicate is encoded into the [Transaction Access List](https://eips.e
 
 Therefore, we use the [Predicate Utils](../../../utils/predicate/README.md) package to encode the actual byte slice of size N into the access list.
 
+### Performance Optimization: C-Chain to Subnet
+
+To support C-Chain to Subnet communication, or more generally Primary Network to Subnet communication, we special case the C-Chain for two reasons:
+
+1. Every Subnet validator validates the C-Chain
+2. The Primary Network has the largest possible number of validators
+
+Since the Primary Network has the largest possible number of validators for any Subnet on Avalanche, it would also be the most expensive Subnet to verify Avalanche Warp Messages from (most signatures/public keys required to verify it). Luckily, we can do something much smarter.
+
+When a Subnet receives a message from a blockchain on the Primary Network, we use the validator set of the receiving Subnet instead of the entire network when validating the message. This means that the C-Chain sending a message can be the exact same as Subnet to Subnet communciation.
```

Comment:
```suggestion
When a Subnet receives a message from a blockchain on the Primary Network, we use the validator set of the receiving Subnet instead of the entire network when validating the message to define a threshold of stake necessary for the receiving network to consider processing the message. This means that the C-Chain sending a message can be the exact same as Subnet to Subnet communication.
```

Reply:
Fixing the typo, but leaving the rest because I think it makes it more difficult to read.